### PR TITLE
chore: update to TypeScript 5

### PR DESCRIPTION
**Pull request by gre**
### 📝 Description

update to TypeScript v5.

### ❓ Context

- **Impacted projects**: `*` 
- **Linked resource(s)**: https://ledgerhq.atlassian.net/browse/LIVE-7641 

### ✅ Checklist

- [x] **Test coverage** 
- [x] **Atomic delivery** 
- [x] **No breaking changes** 

### 🚀 Expectations to reach

_Please make sure you follow these [**Important Steps**](https://github.com/LedgerHQ/ledger-live/blob/develop/CONTRIBUTING.md#important-steps)._

_Pull Requests must pass the CI and be internally validated in order to be merged._

### CODE REVIEW

File: apps/ledger-live-mobile/src/navigation/DeeplinksProvider.tsx [hunk 1]
```diff
@@ -257,15 +257,19 @@ const linkingOptions = {
               [ScreenName.ReceiveSelectCrypto]: "receive",
             },
           },
+          // FIXME: this code was getting overriden. is it the correct one?
```
Comment (gre) [inline apps/ledger-live-mobile/src/navigation/DeeplinksProvider.tsx:260]: @mlegall-ledger can you check which of the 2 codes are to keep? there were twice the definition of `[NavigatorName.Swap]` and i'm not sure which one is to keep.

thanks

File: apps/ledger-live-mobile/src/components/FloatingDebugButton.tsx [hunk 2]
```diff
@@ -88,6 +88,7 @@ const FloatingDebugButton: React.FC<Props> = ({
           {React.isValidElement(Icon) ? (
             Icon
           ) : (
+            /* @ts-expect-error TS 5 can't seem to be able to prove this is a react comopnent here */
```
Comment (gre) [inline apps/ledger-live-mobile/src/components/FloatingDebugButton.tsx:91]: I don't know how to prove to TypeScript that is still valid. if no one knows we will go with the present tradeoff :'(

File: libs/ui/packages/native/.storybook/main.js [hunk 3]
```diff
@@ -9,4 +9,7 @@ module.exports = {
     // "@storybook/addon-ondevice-backgrounds",
     "@storybook/addon-ondevice-notes",
   ],
+  typescript: {
+    reactDocgen: false, // FIXME: this is disabled for now due to incompatibilities with TS 5. re-enable when upgrading storybook.
```
Comment (gre) [inline libs/ui/packages/native/.storybook/main.js:13]: @aboissiere-ledger we will need to see what to do with storybook in future / have a team on it.

File: tools/github-bot/src/features/upToDate/index.ts [hunk 4]
```diff
@@ -64,6 +64,7 @@ export function upToDate(app: Probot) {
     }
   );
 
+  /** @ts-expect-error it seems TypeScript have issues inferring the types here... */
```
Comment (gre) [inline tools/github-bot/src/features/upToDate/index.ts:67]: @valpinkman i have no clue about this part.

Comment (valpinkman) [reply tools/github-bot/src/features/upToDate/index.ts:67]: Yup forget about it it's not important for this part of the code

**Review by valpinkman — APPROVED** [review]

**Pull request merged**